third_party/move/move-vm/types/src/value_serde.rs

```diff
@@ -172,6 +174,112 @@ pub fn serialized_size_allowing_delayed_values(
     })
 }
 
+/// Count number of types constant_serialized_size would visit, used for gas charging.
+/// This is different done type.num_nodes(), as some types are not traversed (i.e. vector),
+/// and for structs types and number of fields matter as well.
+///
+/// Unclear if type_visit_count would be the same for other usages
+/// (for example, whether vector types need to be traversed),
+/// so name it very specifically, and on future usages see how it generalizes.
+pub fn type_visit_count_for_constant_serialized_size(ty_layout: &MoveTypeLayout) -> u64 {
+    match ty_layout {
+        MoveTypeLayout::Bool
+        | MoveTypeLayout::U8
+        | MoveTypeLayout::U16
+        | MoveTypeLayout::U32
+        | MoveTypeLayout::U128
+        | MoveTypeLayout::U256
+        | MoveTypeLayout::U64
+        | MoveTypeLayout::Address
+        | MoveTypeLayout::Signer => 1,
```

Comment:
Should we return None for signer as it shouldn't have been serialized?

Reply:
this is for gas charging. below we do say signer doesn't have constant size.